src/aiidalab_qe/common/mixins.py

```
@@ -71,21 +73,22 @@ def _link_model(self, model: T):
                 (model, "blockers"),
                 (self, "blockers"),
             )
-        for dependency in model.dependencies:
-            dependency_parts = dependency.rsplit(".", 1)
-            if len(dependency_parts) == 1:  # from parent
-                target_model = self
-                trait = dependency
-            else:  # from sibling
-                sibling, trait = dependency_parts
-                target_model = self.get_model(sibling)
-            tl.dlink(
-                (target_model, trait),
-                (model, trait),
-            )
-
-
-class HasProcess(tl.HasTraits):
+        if isinstance(model, Model):
```

Is this check necessary? What if it is not a model, should any error be raised?

---

Yeah, I should handle it differently. There is no need for the check because the model should be typed correctly as a Model. It is typed as a generic T. Maybe I can bound it to Model somehow. But yes, in principle, since this is Python, one can pass whatever they want. We should raise an error if they pass something that is not a model.